.github/workflows/CheckBadMerge.groovy

```diff
@@ -0,0 +1,116 @@
+import java.util.concurrent.Callable
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+
+/**
+ * See https://github.com/gradle/gradle-private/issues/3919
+ *
+ * When merging `releaseX` branch into `master`, we should only use the release note from the `master` branch,
+ * but sometimes changes on release notes.md was brought to master and merged unnoticed,
+ * e.g https://github.com/gradle/gradle/pull/25825
+ *
+ * This script is to check if there is any merge commit that brings changes from release branch to master.
+ * Usage: groovy CheckBadMerge.groovy <commit1> <commit2> ...
+ * If any "bad" merge commit is found, it will print the details and exit with non-zero code.
+ */
+class CheckBadMerge {
+    private static final THREAD_POOL = Executors.newCachedThreadPool()
+
+    private static final List<String> MONITORED_FILES = [
+        "subprojects/docs/src/docs/release/notes.md"
```

Comment:
🤔 Should this also check `subprojects/launcher/src/main/resources/release-features.txt`?

And the new locations of all files presumably: `platforms/documentation/docs/src/docs/release/notes.md` and `platforms/core-runtime/launcher/src/main/resources/release-features.txt`